hax/hax/ha/events.py

```diff
@@ -145,3 +152,78 @@ def ack(self):
            that message using ack method
         """
         self.consumer.ack()
+
+
+class EventProducer():
+    """
+    Module responsible for publishing the event to message bus
+    """
+
+    ADMIN_ID = "hare_admin"
+    __instance = None
+
+    @staticmethod
+    def get_instance():
+        """
+        Static method to fetch the current instance.
+        Performs initialization related to common database and message bus
+        """
+        if not EventProducer.__instance:
+            EventProducer()
+        return EventProducer.__instance
+
+    def __init__(self):
+        logging.debug('Inside EventProducer')
+
+        if EventProducer.__instance is None:
+            EventProducer.__instance = self
+        else:
+            raise Exception("EventProducer is singleton class,"
+                            " use EventProducer.get_instance().")
+
+        message_type = FAULT_TOLERANCE_KEYS.HARE_HA_MESSAGE_TYPE.value
+        self._register_producer(message_type)
+
+        self.producer = MessageProducer(producer_id=HEALTH_EVENT_SOURCES.HARE,
+                                        message_type=message_type)
+
+    def _register_producer(self, message_type,
```

Comment:
_register_producer() must take an instance of producer or please rename the function appropriately.

Reply:
Done